src/vs/workbench/contrib/searchEditor/browser/searchEditor.contribution.ts

```diff
@@ -538,3 +540,36 @@ registerAction2(class OpenSearchEditorAction extends Action2 {
 	}
 });
 //#endregion
+
+//#region Search Editor Working Copy Editor Handler
+class SearchEditorWorkingCopyEditorHandler extends Disposable implements IWorkbenchContribution {
+
+	constructor(
+		@IInstantiationService private readonly instantiationService: IInstantiationService,
+		@IWorkingCopyEditorService private readonly workingCopyEditorService: IWorkingCopyEditorService,
+	) {
+		super();
+
+		this.installHandler();
+	}
+
+	private installHandler(): void {
+		this._register(this.workingCopyEditorService.registerHandler({
+			handles: async workingCopy => {
+				return workingCopy.resource.scheme === SearchEditorConstants.SearchEditorScheme;
+			},
+			isOpen: async (workingCopy, editor) => {
+				return editor instanceof SearchEditorInput && isEqual(workingCopy.resource, editor.modelUri);
+			},
+			createEditor: async workingCopy => {
+				const input = this.instantiationService.invokeFunction(getOrMakeSearchEditorInput, { modelUri: workingCopy.resource, config: {} });
```

Comment:
I am not entirely sure this does the right thing. I need to call `input.setDirty(true)` for the case of a saved search editor restoring. This entire `createEditor` method will only be called if you start VSCode from the command line pointing to a folder and to a file inside that folder (e.g. `code <folder> <folder/file>`) because then we do not restore the editor layout.